dask_cuda/dask_cuda_worker.py

```diff
@@ -0,0 +1,268 @@
+from distributed.config import config
+from distributed.utils import get_ip_interface, parse_timedelta
+from distributed.worker import _ncores
+from distributed.security import Security
+from distributed.cli.utils import (check_python_3, uri_from_host_port,
+                                   install_signal_handlers)
+from distributed.comm import get_address_host_port
+from distributed.preloading import validate_preload_argv
+from distributed.proctitle import (enable_proctitle_on_children,
+                                   enable_proctitle_on_current)
+
+from .local_cuda_cluster import cuda_visible_devices
+
+from toolz import valmap
+from tornado.ioloop import IOLoop, TimeoutError
+from tornado import gen
+
+logger = logging.getLogger('distributed.dask_worker')
+
+
+pem_file_option_type = click.Path(exists=True, resolve_path=True)
+
+
+@click.command(context_settings=dict(ignore_unknown_options=True))
+@click.argument('scheduler', type=str, required=False)
+@click.option('--tls-ca-file', type=pem_file_option_type, default=None,
+              help="CA cert(s) file for TLS (in PEM format)")
+@click.option('--tls-cert', type=pem_file_option_type, default=None,
+              help="certificate file for TLS (in PEM format)")
+@click.option('--tls-key', type=pem_file_option_type, default=None,
+              help="private key file for TLS (in PEM format)")
+@click.option('--worker-port', type=int, default=0,
+              help="Serving computation port, defaults to random")
+@click.option('--nanny-port', type=int, default=0,
+              help="Serving nanny port, defaults to random")
+@click.option('--bokeh-port', type=int, default=0,
+              help="Bokeh port, defaults to random port")
+@click.option('--bokeh/--no-bokeh', 'bokeh', default=True, show_default=True,
+              required=False, help="Launch Bokeh Web UI")
+@click.option('--listen-address', type=str, default=None,
+        help="The address to which the worker binds. "
+             "Example: tcp://0.0.0.0:9000")
+@click.option('--contact-address', type=str, default=None,
+        help="The address the worker advertises to the scheduler for "
+             "communication with it and other workers. "
+             "Example: tcp://127.0.0.1:9000")
+@click.option('--host', type=str, default=None,
+              help="Serving host. Should be an ip address that is"
+                   " visible to the scheduler and other workers. "
+                   "See --listen-address and --contact-address if you "
+                   "need different listen and contact addresses. "
+                   "See --interface.")
+@click.option('--interface', type=str, default=None,
+              help="Network interface like 'eth0' or 'ib0'")
+@click.option('--nthreads', type=int, default=0,
+              help="Number of threads per process.")
+@click.option('--nprocs', type=int, default=1,
+              help="Number of worker processes to launch.  Defaults to one.")
```

Comment:
I suggest that for now we remove this option and define `nprocs` ourselves as the number of available GPUs.  I think that the ideal  current behavior of `dask-cuda-worker` on a DGX would be to create eight processes, unless restricted by `CUDA_VISIBLE_DEVICES`.  

The function `utils.py::get_n_gpus` may be useful.  I think that this logic is also already in LocalCUDACluster, which may serve as a reference.

Reply:
I'd like t keep this open for the non-1:1 setups i described in https://github.com/rapidsai/dask-cuda/issues/3#issuecomment-461190475 where we have multiple GPUs per process.

Reply:
Many flags interact with nprocs flag, mostly ones that can't be used with it. If we remove the flag and internally set nprocs = #GPUs, then those flags are never usable and things unravel a bit. How about leaving the flag but defaulting it to #GPUs instead of 1?

Reply:
Most of the other flags we'll also want to remove.  The nanny option will go away (we need to have a nanny regardless).  And the threads option will also go away (we need to have one thread per gpu until we figure out a nice concurrency story for threads on GPUs)

Reply:
@beberg mentioned that other parts of Dask may expect keywords like `--nthreads` to exist.  Indeed after some looking into things it looks like dask-jobqueue does expect an `--nthreads` keyword.

Reply:
I still think that we should drop the `--nprocs` keyword (this is infrequently used in deployment solutions), and probably also the `--nanny` keyword.  If we keep either of those then they should probably err if the user tries to use them.  I'm inclined to remove though.

Reply:
Dropped
-    worker_port,
-    listen_address,
-    contact_address,
-    nanny_port,
-    nprocs,
-    nanny,

nprocs internally set to #GPUs